### PR TITLE
pkg: kernel: Sign ZFS modules after strip

### PR DESCRIPTION
**Pull request by rene**
In order to save some disk space, all modules are stripped when building kernel for ARM64. The modules are signed again after the strip, but ZFS modules are not being signed.

This commit fixes this issue and some comments along the Dockerfile.

### CODE REVIEW

**Review by eriknordmark — COMMENTED** [review]
Without this fix do we fail to load the zfs module?

**Comment by zedi-pramodh** [comment]
Just curious if it causes any upgrade issues ?

**Comment by shjala** [comment]
> Just curious if it causes any upgrade issues ?

Happens only an arm64.

File: pkg/kernel/Dockerfile [hunk 1]
```diff
@@ -290,9 +290,14 @@ RUN KERNEL_DEF_CONF="/linux/arch/${KERNEL_ARCH}/configs/${KERNEL_DEFCONFIG}"; \
         SIG_KEY_SRCPREFIX=$(sed -n '/^MODULE_SIG_KEY_SRCPREFIX=/s///p' ${KERNEL_DEF_CONF} | tr -d '"') ; \
         SIG_KEY=$(sed -n '/^CONFIG_MODULE_SIG_KEY=/s///p' ${KERNEL_DEF_CONF} | tr -d '"') ; \
         scripts/sign-file "${SIG_HASH}" "/linux/${SIG_KEY_SRCPREFIX}${SIG_KEY}" /linux/certs/signing_key.x509 $(find /tmp/kernel-modules/lib/modules -name 8821cu.ko); \
+        if [ "${EVE_TARGET_ARCH}" = aarch64 ];then \
```
Comment (rouming) [inline pkg/kernel/Dockerfile:293]: Maybe a small comment explaining why is arm specific?

**Review by rouming — APPROVED** [review]
Ack

**Comment by shjala** [comment]
Auto signing happens for `modules_install` targets and not `install`, my bad.
@rene I see the same issue exist on new-kernel too, would you please add commit to fix that too?

**Comment by rene** [comment]
> Without this fix do we fail to load the zfs module?

Yes @eriknordmark , ZFS cannot be loaded on ARM.....

**Comment by rene** [comment]
> Just curious if it causes any upgrade issues ?

It's only affecting ARM @zedi-pramodh , fortunately we don't see so many ZFS installations on ARM, but if the device uses ZFS, then yes, it will affect the upgrade...

**Comment by rene** [comment]
> ~Auto signing happens for `modules_install` targets and not `install`~, my bad. @rene I see the same issue exist on new-kernel too, would you please add commit to fix that too?

thanks for pointed out @shjala , I will fix new-kernel as well...

**Comment by rene** [comment]
Updates in this PR:
- Fix reviewer's comments
- Fix new-kernel module stripping
- Fix ZFS signing for new-kernel as well

File: pkg/new-kernel/Dockerfile [hunk 2]
```diff
@@ -272,13 +272,18 @@ RUN if [ "${EVE_TARGET_ARCH}" != riscv64 ]; then \
             SIG_KEY_SRCPREFIX=$(sed -n '/^MODULE_SIG_KEY_SRCPREFIX=/s///p' ${KERNEL_DEF_CONF} | tr -d '"') ; \
             SIG_KEY=$(sed -n '/^CONFIG_MODULE_SIG_KEY=/s///p' ${KERNEL_DEF_CONF} | tr -d '"') ; \
             scripts/sign-file "${SIG_HASH}" "/linux/${SIG_KEY_SRCPREFIX}${SIG_KEY}" /linux/certs/signing_key.x509 $(find /tmp/kernel-modules/lib/modules -name 8821cu.ko); \
+            if [ "${EVE_TARGET_ARCH}" = aarch64 ];then \
```
Comment (shjala) [inline pkg/new-kernel/Dockerfile:275]: we have zfs on riscv64 too, only 8821cu is not installed on riscv64.

Comment (rene) [reply pkg/new-kernel/Dockerfile:275]: nice catch, that's need to be different for new-kernel, I will fix...

**Comment by rene** [comment]
Updates in this PR:
- Fix architecture checks for new-kernel

**Pull request merged**